client/landing/stepper/declarative-flow/internals/steps-repository/unified-plans/index.tsx

```diff
@@ -158,40 +134,6 @@ const PlansStepAdaptor: StepType< {
 		setPlanInterval( intervalType );
 	};
 
-	/**
-	 *  Plan step switches the selected theme to default twentytwentyfour when the plan is free
-	 *  but the selected design requires a paid plan.
-	 */
-	const switchPaidDesignToDefault = ( stepInfo: ProvidedDependencies ) => {
-		const hasPaidDesign = Boolean(
-			selectedDesign?.design_tier && selectedDesign?.design_tier !== 'free'
-		);
-		const isOnboarding = isOnboardingFlow( props.flow ) && isGoalFirstExperiment;
```

Comment:
Because `isGoalFirstExperiment` is always `false`, the `isOnboarding` variable is always `false`, the whole `switchPaidDesignToDefault` function will always `return` early.

Reply:
Nice find. Can we get someone to confirm that the pre-existing logic is intentional and still alright after we remove this?